csharp/extractor/Semmle.Extraction.CSharp.DependencyFetching/DependencyManager.cs

```diff
@@ -723,7 +665,7 @@ private void AnalyseProject(FileInfo project)
             }
         }
 
-        public void Dispose(TemporaryDirectory? dir, string name)
+        public static void DisposeTempDirectory(TemporaryDirectory? dir, string name, ILogger logger)
```

Comment:
nit: Maybe consider to put this method in another (static class) or incorporate the try/catch logic directly in the `Dispose` method in the `TemporaryDirectory` class.